bot/plotting/comparison_generator.py

```diff
@@ -127,15 +130,36 @@ def comparison_generator(
 
                 is_experiment = False
 
+                # vary "Current function [A]" and "Ambient temperature [K]"
+                # if comparing models with a constant discharge
+                if number_of_comp != 1:
+                    params, min_param_value = parameter_value_generator(
+                        parameter_values.copy(), "Current function [A]"
+                    )
+                    (
+                        final_params,
+                        varied_value_temp
+                    ) = parameter_value_generator(
+                        params.copy(),
+                        "Ambient temperature [K]",
+                        lower_bound=265,
+                        upper_bound=355
+                    )
+                    parameter_values_for_comp = dict(
+                        list(enumerate([final_params]))
+                    )
+
                 batch_study = pybamm.BatchStudy(
                     models=models_for_comp,
                     parameter_values=parameter_values_for_comp,
                     permutations=True,
                 )
 
                 # if "Current function [A]" is varied, change the t_end
-                if param_to_vary == "Current function [A]":
-                    factor = min_param_value / varied_value
+                if min_param_value != 100:
+                    factor = min_param_value / parameter_values[
```

Comment:
instead of calculating `min_param_value` as you loop over changing the parameters, you could just calculate the smallest current here from the dictionary of `parameter_values_for_comp`